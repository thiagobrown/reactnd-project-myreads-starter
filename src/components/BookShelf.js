import React from 'react'
import PropTypes from 'prop-types'

import Book from './Book'

const BookShelf = ({ title, books = [], onUpdateBook }) => {
     return (
        <div className="bookshelf">
            <h2 className="bookshelf-title">{title}</h2>
            <div className="bookshelf-books">
                <ol className="books-grid">
                    {books.map(b => (
                        <Book key={b.id} book={b} onUpdateBook={onUpdateBook} />
                    ))}
                </ol>
            </div>
        </div>
    )
}

BookShelf.propTypes = {
    title: PropTypes.string.isRequired,
    books: PropTypes.array,
    onUpdateBook: PropTypes.func.isRequired
}

export default BookShelf