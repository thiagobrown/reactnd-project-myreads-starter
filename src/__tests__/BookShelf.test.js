import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import BookShelf from './../components/BookShelf'

describe('Testing BookShelf Component', () => {
    let props = {
        title: '',
        books: [],
        onUpdateBook: jest.fn()
    };

    const wrapper = shallow(<BookShelf {...props} />);

    it('render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('check render when state changes', () => {
        wrapper.setProps(
            {
                title: 'read',
                books: [{
                    id: 1,
                    title: 'Book One',
                    authors: ['Authors One'],
                    imageLinks: { thumbnail: 'url/book/one' },
                    shelf: 'read'
                }],
                onUpdateBook: jest.fn()
            }
        );
        expect(wrapper).toMatchSnapshot();
    });
});