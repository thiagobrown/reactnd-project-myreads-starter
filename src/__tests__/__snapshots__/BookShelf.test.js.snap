exports[`Testing BookShelf Component check render when state changes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookShelf
    books={
        Array [
            Object {
              "authors": Array [
                "Authors One",
              ],
              "id": 1,
              "imageLinks": Object {
                "thumbnail": "url/book/one",
              },
              "shelf": "read",
              "title": "Book One",
            },
          ]
    }
    onUpdateBook={[Function]}
    title="read" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="bookshelf-title">
          read
</h2>,
        <div
          className="bookshelf-books">
          <ol
                    className="books-grid">
                    <Book
                              book={
                                        Object {
                                                  "authors": Array [
                                                    "Authors One",
                                                  ],
                                                  "id": 1,
                                                  "imageLinks": Object {
                                                    "thumbnail": "url/book/one",
                                                  },
                                                  "shelf": "read",
                                                  "title": "Book One",
                                                }
                              }
                              onUpdateBook={[Function]} />
          </ol>
</div>,
      ],
      "className": "bookshelf",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "read",
          "className": "bookshelf-title",
        },
        "ref": null,
        "rendered": "read",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ol
            className="books-grid">
            <Book
                        book={
                                    Object {
                                                "authors": Array [
                                                  "Authors One",
                                                ],
                                                "id": 1,
                                                "imageLinks": Object {
                                                  "thumbnail": "url/book/one",
                                                },
                                                "shelf": "read",
                                                "title": "Book One",
                                              }
                        }
                        onUpdateBook={[Function]} />
</ol>,
          "className": "bookshelf-books",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Book
                book={
                                Object {
                                                "authors": Array [
                                                  "Authors One",
                                                ],
                                                "id": 1,
                                                "imageLinks": Object {
                                                  "thumbnail": "url/book/one",
                                                },
                                                "shelf": "read",
                                                "title": "Book One",
                                              }
                }
                onUpdateBook={[Function]} />,
            ],
            "className": "books-grid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "book": Object {
                  "authors": Array [
                    "Authors One",
                  ],
                  "id": 1,
                  "imageLinks": Object {
                    "thumbnail": "url/book/one",
                  },
                  "shelf": "read",
                  "title": "Book One",
                },
                "onUpdateBook": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ol",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="bookshelf-title">
            read
</h2>,
          <div
            className="bookshelf-books">
            <ol
                        className="books-grid">
                        <Book
                                    book={
                                                Object {
                                                            "authors": Array [
                                                              "Authors One",
                                                            ],
                                                            "id": 1,
                                                            "imageLinks": Object {
                                                              "thumbnail": "url/book/one",
                                                            },
                                                            "shelf": "read",
                                                            "title": "Book One",
                                                          }
                                    }
                                    onUpdateBook={[Function]} />
            </ol>
</div>,
        ],
        "className": "bookshelf",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "read",
            "className": "bookshelf-title",
          },
          "ref": null,
          "rendered": "read",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ol
              className="books-grid">
              <Book
                            book={
                                          Object {
                                                        "authors": Array [
                                                          "Authors One",
                                                        ],
                                                        "id": 1,
                                                        "imageLinks": Object {
                                                          "thumbnail": "url/book/one",
                                                        },
                                                        "shelf": "read",
                                                        "title": "Book One",
                                                      }
                            }
                            onUpdateBook={[Function]} />
</ol>,
            "className": "bookshelf-books",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Book
                  book={
                                    Object {
                                                      "authors": Array [
                                                        "Authors One",
                                                      ],
                                                      "id": 1,
                                                      "imageLinks": Object {
                                                        "thumbnail": "url/book/one",
                                                      },
                                                      "shelf": "read",
                                                      "title": "Book One",
                                                    }
                  }
                  onUpdateBook={[Function]} />,
              ],
              "className": "books-grid",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "book": Object {
                    "authors": Array [
                      "Authors One",
                    ],
                    "id": 1,
                    "imageLinks": Object {
                      "thumbnail": "url/book/one",
                    },
                    "shelf": "read",
                    "title": "Book One",
                  },
                  "onUpdateBook": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ol",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Testing BookShelf Component render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookShelf
    books={Array []}
    onUpdateBook={[Function]}
    title="" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="bookshelf-title" />,
        <div
          className="bookshelf-books">
          <ol
                    className="books-grid">
                    
          </ol>
</div>,
      ],
      "className": "bookshelf",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "bookshelf-title",
        },
        "ref": null,
        "rendered": "",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ol
            className="books-grid">
            
</ol>,
          "className": "bookshelf-books",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "books-grid",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ol",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="bookshelf-title" />,
          <div
            className="bookshelf-books">
            <ol
                        className="books-grid">
                        
            </ol>
</div>,
        ],
        "className": "bookshelf",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "bookshelf-title",
          },
          "ref": null,
          "rendered": "",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ol
              className="books-grid">
              
</ol>,
            "className": "bookshelf-books",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "books-grid",
            },
            "ref": null,
            "rendered": Array [],
            "type": "ol",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Testing BookShelf Component verify render when state changes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookShelf
    books={
        Array [
            Object {
              "authors": Array [
                "Authors One",
              ],
              "id": 1,
              "imageLinks": Object {
                "thumbnail": "url/book/one",
              },
              "shelf": "read",
              "title": "Book One",
            },
          ]
    }
    onUpdateBook={[Function]}
    title="read" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="bookshelf-title">
          read
</h2>,
        <div
          className="bookshelf-books">
          <ol
                    className="books-grid">
                    <Book
                              book={
                                        Object {
                                                  "authors": Array [
                                                    "Authors One",
                                                  ],
                                                  "id": 1,
                                                  "imageLinks": Object {
                                                    "thumbnail": "url/book/one",
                                                  },
                                                  "shelf": "read",
                                                  "title": "Book One",
                                                }
                              }
                              onUpdateBook={[Function]} />
          </ol>
</div>,
      ],
      "className": "bookshelf",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "read",
          "className": "bookshelf-title",
        },
        "ref": null,
        "rendered": "read",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ol
            className="books-grid">
            <Book
                        book={
                                    Object {
                                                "authors": Array [
                                                  "Authors One",
                                                ],
                                                "id": 1,
                                                "imageLinks": Object {
                                                  "thumbnail": "url/book/one",
                                                },
                                                "shelf": "read",
                                                "title": "Book One",
                                              }
                        }
                        onUpdateBook={[Function]} />
</ol>,
          "className": "bookshelf-books",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Book
                book={
                                Object {
                                                "authors": Array [
                                                  "Authors One",
                                                ],
                                                "id": 1,
                                                "imageLinks": Object {
                                                  "thumbnail": "url/book/one",
                                                },
                                                "shelf": "read",
                                                "title": "Book One",
                                              }
                }
                onUpdateBook={[Function]} />,
            ],
            "className": "books-grid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "book": Object {
                  "authors": Array [
                    "Authors One",
                  ],
                  "id": 1,
                  "imageLinks": Object {
                    "thumbnail": "url/book/one",
                  },
                  "shelf": "read",
                  "title": "Book One",
                },
                "onUpdateBook": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ol",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="bookshelf-title">
            read
</h2>,
          <div
            className="bookshelf-books">
            <ol
                        className="books-grid">
                        <Book
                                    book={
                                                Object {
                                                            "authors": Array [
                                                              "Authors One",
                                                            ],
                                                            "id": 1,
                                                            "imageLinks": Object {
                                                              "thumbnail": "url/book/one",
                                                            },
                                                            "shelf": "read",
                                                            "title": "Book One",
                                                          }
                                    }
                                    onUpdateBook={[Function]} />
            </ol>
</div>,
        ],
        "className": "bookshelf",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "read",
            "className": "bookshelf-title",
          },
          "ref": null,
          "rendered": "read",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ol
              className="books-grid">
              <Book
                            book={
                                          Object {
                                                        "authors": Array [
                                                          "Authors One",
                                                        ],
                                                        "id": 1,
                                                        "imageLinks": Object {
                                                          "thumbnail": "url/book/one",
                                                        },
                                                        "shelf": "read",
                                                        "title": "Book One",
                                                      }
                            }
                            onUpdateBook={[Function]} />
</ol>,
            "className": "bookshelf-books",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Book
                  book={
                                    Object {
                                                      "authors": Array [
                                                        "Authors One",
                                                      ],
                                                      "id": 1,
                                                      "imageLinks": Object {
                                                        "thumbnail": "url/book/one",
                                                      },
                                                      "shelf": "read",
                                                      "title": "Book One",
                                                    }
                  }
                  onUpdateBook={[Function]} />,
              ],
              "className": "books-grid",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "book": Object {
                    "authors": Array [
                      "Authors One",
                    ],
                    "id": 1,
                    "imageLinks": Object {
                      "thumbnail": "url/book/one",
                    },
                    "shelf": "read",
                    "title": "Book One",
                  },
                  "onUpdateBook": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ol",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
